python/hsfs/engine/python.py

```diff
@@ -1228,39 +1292,59 @@ def add_file(self, file: Optional[str]) -> Optional[str]:
 
     def _apply_transformation_function(
         self,
-        transformation_functions: Dict[
-            str, transformation_function_attached.TransformationFunctionAttached
-        ],
+        transformation_functions: List[transformation_function.TransformationFunction],
         dataset: Union[pd.DataFrame, pl.DataFrame],
     ) -> Union[pd.DataFrame, pl.DataFrame]:
-        for (
-            feature_name,
-            transformation_fn,
-        ) in transformation_functions.items():
-            if isinstance(dataset, pl.DataFrame) or isinstance(
-                dataset, pl.dataframe.frame.DataFrame
-            ):
-                dataset = dataset.with_columns(
-                    pl.col(feature_name).map_elements(
-                        transformation_fn.transformation_fn
-                    )
-                )
+        """
+        Apply transformation function to the dataframe.
+
+        # Arguments
+            transformation_functions `List[transformation_function.TransformationFunction]` : List of transformation functions.
+            dataset `Union[pd.DataFrame, pl.DataFrame]`: A pandas or polars dataframe.
+        # Returns
+            `DataFrame`: A pandas dataframe with the transformed data.
+        # Raises
+            `FeatureStoreException`: If any of the features mentioned in the transformation function is not present in the Feature View.
+        """
+        transformed_features = set()
+
+        if isinstance(dataset, pl.DataFrame) or isinstance(
+            dataset, pl.dataframe.frame.DataFrame
+        ):
+            # Converting polars dataframe to pandas because currently we support only pandas UDF's as transformation functions.
+            if os.getenv("USE_PYARROW_EXTENSION", False):
```

Comment:
seems not necessary to use env var here, just a method is sufficient as you do not expect user to set it.

Reply:
Changed to a function.